recipe/meta.yaml

```diff
@@ -159,55 +155,84 @@ outputs:
         - sqlite
         - zlib
         # requirements specified by the package itself
-        - absl-py >=0.10,<1
-        - astunparse >=1.6.3,<1.7
-        - python-flatbuffers >=1.12,<2
-        - google-pasta >=0.2,<1
-        - h5py >=3.1.0,<3.2
-        - keras-preprocessing >=1.1.2,<1.2
-        - numpy >=1.19.2,<1.20
-        - opt_einsum >=3.3,<3.4
+        - absl-py >=0.4.0
+        - astunparse >=1.6.0
+        - gast >=0.2.1,<0.5.0
+        - google-pasta >=0.1.1
+        - h5py >=2.9.0
+        - keras-preprocessing >=1.1.1
+        - numpy
+        - opt_einsum >=2.3.2
         - protobuf >=3.9.2
-        - six >=1.15,<1.16
-        - termcolor >=1.1,<1.2
-        - typing_extensions >=3.7.4,<3.8
-        - wheel >=0.35,<1
-        - wrapt >=1.12.1,<1.13
-        # upstream wants exact pin for gast
-        - gast ==0.4.0
+        - python-flatbuffers >=1.12,<3
+        - six >=1.12
+        - termcolor >=1.1.0
+        - typing_extensions >=3.6.6
+        - wrapt >=1.11.0
         # TF-API needs to move in sync
+        # XXX: actually, TF 2.7.0 still demands tb 2.6
         - tensorboard >=2.6,<2.7
-        - keras >=2.6,<2.7
+        - keras >=2.7,<2.8
       run:
         - python
-        - absl-py >=0.10,<1
-        - astunparse >=1.6.3,<1.7
-        - python-flatbuffers >=1.12,<2
-        - google-pasta >=0.2,<1
+        - absl-py >=0.4.0
+        - astunparse >=1.6.0
+        - gast >=0.2.1,<0.5.0
+        - google-pasta >=0.1.1
         - grpcio {{ grpc_cpp }}.*
-        - h5py >=3.1.0,<3.2
-        - keras-preprocessing >=1.1.2,<1.2
-        - numpy >=1.19.2,<1.20
-        - opt_einsum >=3.3,<3.4
+        - h5py >=2.9.0
+        - keras-preprocessing >=1.1.1
+        - {{ pin_compatible('numpy') }}
+        - opt_einsum >=2.3.2
         - protobuf >=3.9.2
-        - six >=1.15,<1.16
-        - termcolor >=1.1,<1.2
-        - typing_extensions >=3.7.4,<3.8
-        - wheel >=0.35,<1
-        - wrapt >=1.12.1,<1.13
-        # upstream wants exact pin for gast
-        - gast ==0.4.0
+        - python-flatbuffers >=1.12,<3
+        - six >=1.12
+        - termcolor >=1.1.0
+        - typing_extensions >=3.6.6
+        - wrapt >=1.11.0
         # TF-API needs to move in sync
+        # XXX: actually, TF 2.7.0 still demands tb 2.6
         - tensorboard >=2.6,<2.7
-        - keras >=2.6,<2.7
+        - keras >=2.7,<2.8
+        # avoid that people without GPUs needlessly download ~0.5-1GB
+        - __cuda  # [cuda_compiler_version != "None"]
+    # TODO: decide on the name of the package
+    # run_constrained:
+    #   What is the difference between these two packages?
+    #   - tensorflow-io-gcs-filesystem >= 0.21.0
+    #   - tensorflow-io >=0.21.0
+
+  # compatibility output wrapping core package, since tensorflow-base
+  # was used as an output before 2.7 (should we add a deprecation warning?)
+  - name: tensorflow
```

Comment:
Is this really deprecated? From the looks of the diff, we still need it.

Reply:
This might be a remnant of @hmaarrfk taking over my changes. Seems the comment up top was updated, but not this one.

Reply:
Yes sorry for leaving this lingering.

It is not deprecated.

I removed the comment.